src/EventGrid/EventGrid/Domain/GetAzureEventGridDomainTopic.cs

```diff
@@ -0,0 +1,168 @@
+            HelpMessage = EventGridConstants.DomainTopicNameHelp,
+            ParameterSetName = DomainTopicNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/domains/topics", nameof(ResourceGroupName), nameof(DomainName))]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasDomainTopicName)]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainOrDomainTopicResourceIdHelp,
+            ParameterSetName = ResourceIdDomainTopicParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = DomainTopicNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceIdDomainTopicParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ODataQuery { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = DomainTopicNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceIdDomainTopicParameterSet)]
+        [ValidateRange(1, 100)]
+        public int Top { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.NextLinkHelp,
+            ParameterSetName = NextLinkParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string NextLink { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string resourceGroupName = string.Empty;
+            string domainName = string.Empty;
+            string domainTopicName = string.Empty;
+            string newNextLink = null;
+            IEnumerable<DomainTopic> domainTopicsList;
+            List <PSDomainTopicListInstance> psDomainTopicsList = new List<PSDomainTopicListInstance>();
+            int? providedTop = null;
+
+            if (MyInvocation.BoundParameters.ContainsKey(nameof(this.Top)))
+            {
+                providedTop = this.Top;
+            }
+
+            if (!string.IsNullOrEmpty(this.ResourceId))
+            {
+                EventGridUtils.GetResourceGroupNameAndDomainNameAndDomainTopicName(this.ResourceId, out resourceGroupName, out domainName, out domainTopicName);
```

Comment:
@ahamad-MS same comment about changing the variables to the parameters rather than temporary variables and then removing the below if-statement

Reply:
Same here. Cannot change it as proposed.